packages/web/components/QuestChain/UploadProof.tsx

```diff
@@ -185,87 +205,33 @@ export const UploadProof: React.FC<{
         isDisabled={chainId === questChain.chainId}
       >
         <StatusedSubmitButton
+          px={[8, 12]}
+          label="Submit Proof"
           onClick={() => {
-            setIsSubmittingProof(true);
-            onOpen();
+            if (!chainId || chainId !== questChain.chainId || !provider) {
+              addToast({
+                description: `Wrong Chain, please switch to ${
```

Comment:
You're handling 3 separate cases and the description only covers 1, maybe change the description to something like: "Please ensure your wallet is connected to the x chain."